src/Eloquent/Traits/HasAdjacencyList.php

```diff
@@ -436,22 +443,28 @@ public function isIntegerAttribute($attribute)
      * Create a new Eloquent query builder for the model.
      *
      * @param \Illuminate\Database\Query\Builder $query
-     * @return \Staudenmeir\LaravelAdjacencyList\Eloquent\Builder<self>
+     * @return \Staudenmeir\LaravelAdjacencyList\Eloquent\Builder<static>
      */
     public function newEloquentBuilder($query)
     {
-        return new \Staudenmeir\LaravelAdjacencyList\Eloquent\Builder($query);
+        /** @var \Staudenmeir\LaravelAdjacencyList\Eloquent\Builder<static> $builder */
+        $builder = new \Staudenmeir\LaravelAdjacencyList\Eloquent\Builder($query);
+
+        return $builder;
```

Comment:
Just a nit, but I would do what you've been doing:

```suggestion
        // @phpstan-ignore return.type
        return new \Staudenmeir\LaravelAdjacencyList\Eloquent\Builder($query);
```

Reply:
Any idea why the docblock is needed at all / how we can remove it?

Reply:
It can be tricky to get the inner types properly populated when constructing an object sometimes